src/nupic/utils/LoggingException.hpp

```diff
@@ -59,7 +59,7 @@ namespace nupic
     }
 
     // for Index.hpp: // because stringstream cant take << vector
-    LoggingException& operator<<(std::vector<unsigned int, std::allocator<unsigned int> >  v)
+    template <typename T> LoggingException& operator<<(std::vector<T, std::allocator<T> >  v)
```

Comment:
Original doesn't cope with other 64 bit data models (non-Solaris/Linux). The original form of the operator could be duplicated and guarded to handle the different models? or templated like this.
http://stackoverflow.com/questions/3325706/lp64-llp64-and-the-il32-transition